### PR TITLE
made CircularDependencyExcpetion class public

### PR DESCRIPTION
**Pull request by tflynt91**
Closes https://github.com/dotnet/msbuild/issues/5811

### CODE REVIEW

**Comment by BenVillalobos** [comment]
@tflynt91 could you provide context for this change in the description?

**Comment by BenVillalobos** [comment]
`##[error] After building, there are changed files.  Please build locally and include these changes in your pull request.` Can you run `build.cmd` and push up any changes?

**Comment by tflynt91** [comment]
Will do!

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/msbuild?pullRequest=5988) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/msbuild?pullRequest=5988) before we can accept your contribution.<br/><br/>:x: tflynt91 [sign now](https://cla.dotnetfoundation.org/dotnet/msbuild?pullRequest=5988)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/msbuild?pullRequest=5988) it.</sub>

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/msbuild?pullRequest=5988) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/msbuild?pullRequest=5988) before we can accept your contribution.<br/><br/>:x: tflynt91 [sign now](https://cla.dotnetfoundation.org/dotnet/msbuild?pullRequest=5988)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/msbuild?pullRequest=5988) it.</sub>

**Review by BenVillalobos — APPROVED** [review]
The last blocker is to sign the CLA. Otherwise, LGTM!

**Comment by tflynt91** [comment]
great thanks! I believe I signed it but maybe the status is still pending

File: src/Build/BackEnd/Shared/CircularDependencyException.cs [hunk 1]
```diff
@@ -31,7 +31,7 @@ internal CircularDependencyException(string message)
         /// <summary>
         /// Constructor for deserialization.
         /// </summary>
-        protected CircularDependencyException(SerializationInfo info, StreamingContext context)
+        public CircularDependencyException(SerializationInfo info, StreamingContext context)
```
Comment (rainersigwald) [inline src/Build/BackEnd/Shared/CircularDependencyException.cs:34]: Is this change necessary? I think it's ok for this ideally-unused constructor to continue to be protected.

File: ref/Microsoft.Build/net/Microsoft.Build.cs [hunk 2]
```diff
@@ -1,5 +1,12 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
+namespace Microsoft.Build.BackEnd
```
Comment (rainersigwald) [inline ref/Microsoft.Build/net/Microsoft.Build.cs:3]: We haven't exposed this namespace before, so I think we should move the exception into the `Microsoft.Build.Exceptions` namespace where the others are:

https://github.com/dotnet/msbuild/blob/dcf8be378c86c0fff1dea57e0b532f322f09d191/ref/Microsoft.Build/net/Microsoft.Build.cs#L887-L890

Can you change the namespace in `CircularDependencyException.cs`?

**Review by rainersigwald — CHANGES_REQUESTED** [review]
Thanks! Just a couple of nitpicks to address.

I don't know why there are two comments about the CLA but one of them and the check are happy so I believe you signed it :)

File: ref/Microsoft.Build/net/Microsoft.Build.cs (hunk 2 above)
Comment (tflynt91) [reply ref/Microsoft.Build/net/Microsoft.Build.cs]: sounds good! I think keeping it protected was causing some error preventing it from building. The build worked on my end, I think the namespace change might resolve that, so I changed it back to protected.

**Review by rainersigwald — APPROVED** [review]
Looks great. Thank you!
